stubs/paramiko/paramiko/util.pyi

```diff
@@ -44,7 +44,7 @@ def retry_on_signal(function: Callable[[], _T]) -> _T: ...
 def constant_time_bytes_eq(a: AnyStr, b: AnyStr) -> bool: ...
 
 class ClosingContextManager:
-    def __enter__(self: _TC) -> _TC: ...
+    def __enter__(self: Self) -> Self: ...
```

Comment:
This is also semantically different to the existing stub, but looks more correct to me. https://github.com/paramiko/paramiko/blob/88f35a537428e430f7f26eee8026715e357b55d6/paramiko/util.py#L300